compute/list_all_instances.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_instances_list_all]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// listAllInstances gets all instances present in a project, grouped by their zone.
+func listAllInstances(w io.Writer, projectID string) error {
+	// zone := "europe-central2-b"
+	ctx := context.Background()
+	projectsClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return fmt.Errorf("NewInstancesRESTClient: %v", err)
+	}
+
+	defer projectsClient.Close()
+
+	req := &computepb.AggregatedListInstancesRequest{
+		Project: projectID,
+	}
+
+	resp, err := projectsClient.AggregatedList(ctx, req)
```

Comment:
We can include these for now, but I believe this method should be paginated. This is still being worked on right now for go. See https://github.com/googleapis/gapic-generator-go/pull/711. If we add listing samples they will need to be updated in not too long.